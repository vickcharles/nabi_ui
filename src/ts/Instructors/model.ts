export interface ExperienceType {
  id?: number;
  employer: string;
  jobTitle: string;
  fromMonth: string;
  fromYear: string;
  toMonth: string;
  toYear: string;
  location: string;
}

export enum DegreeType {
  associate = 'Associate Degree',
  bachelors = 'Bachelor\'s Degree',
  graduate = 'Graduate Degreee',
  professional = 'Professional Degree',
  certification = 'Certification',
  other = 'Other'
}

export interface EducationType {
  id?: number;
  schoolName: string;
  graduationYear: string;
  degreeType: DegreeType;
  fieldOfStudy: string;
}

export enum PlaceForLessonsOptions {
  home = 'Student\'s home',
  studio = 'Instructor\'s studio',
  online = 'Online',
}

export interface PlaceForLessonsType {
  home: boolean;
  studio: boolean;
  online: boolean;
}

export enum LessonDuration {
  thirtyMins = '30 mins',
  fortyFiveMins = '45 mins',
  sixtyMins = '60 mins',
  ninetyMins = '90 mins'
}

export interface RatesType {
  thirtyMinsRate: number;
  fortyFiveMinsRate: number;
  sixtyMinsRate: number;
  ninetyMinsRate: number;
}

export enum SkillLevel {
  beginner = 'Beginner',
  intermediate = 'Intermediate',
  advanced = 'Advanced'
}

export interface InstrumentsType {
  instrument: string;
  skillLevel: SkillLevel;
}

export interface InstructorType {
  userId: string;
  bio?: string;
  instruments?: InstrumentsType[];
  rates?: RatesType;
  placeForLessons?: PlaceForLessonsType;
  studioAddress?: string;
  education?: EducationType[];
  experience?: ExperienceType[];
}

export type IState = InstructorType[];
