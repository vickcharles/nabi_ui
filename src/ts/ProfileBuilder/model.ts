export interface InstructorState {
  userId: string;
  bio?: string;
}

export type IState = InstructorState[];
  