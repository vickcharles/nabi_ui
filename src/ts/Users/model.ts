export enum HearAboutUs {
  Google = 'google',
  Facebook = 'facebook',
  Craigslist = 'craigslist',
  PrintAds = 'printAds',
  WordOfMouth = 'wordOfMouth',
  CurrentNabiUser = 'currentNabiUser',
  Other= 'other'
}

export interface HearAboutUsInfo {
  value: HearAboutUs;
  label: string;
}

export const hearAboutUsInfo = {
  Google: <HearAboutUsInfo> {
    value: HearAboutUs.Google,
    label: 'Google'
  },
  Facebook: <HearAboutUsInfo> {
    value: HearAboutUs.Facebook,
    label: 'Facebook'
  },
  Craigslist: <HearAboutUsInfo> {
    value: HearAboutUs.Craigslist,
    label: 'Craigslist'
  },
  PrintAds: <HearAboutUsInfo> {
    value: HearAboutUs.PrintAds,
    label: 'Print Ads'
  },
  WordOfMouth: <HearAboutUsInfo> {
    value: HearAboutUs.WordOfMouth,
    label: 'Word Of Mouth'
  },
  CurrentNabiUser: <HearAboutUsInfo> {
    value: HearAboutUs.CurrentNabiUser,
    label: 'Current Nabi User'
  },
  Other: <HearAboutUsInfo> {
    value: HearAboutUs.Other,
    label: 'Other'
  }
};

export enum Role {
  student = 'student',
  instructor = 'instructor',
  parent = 'parent'
}

export interface UserState {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  zipCode: string;
  role: Role;
  hearAboutUs: string;
  avatar?: string;
  displayName: string;
}

export type IState = UserState[];
