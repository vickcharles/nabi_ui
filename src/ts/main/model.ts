/**
 * State for firing redirect
 * @interface RedirectState
 */
export interface RedirectState {
  fireRedirect: boolean;
}
