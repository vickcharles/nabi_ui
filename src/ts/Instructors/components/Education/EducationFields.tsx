import * as React from 'react';

import TextField from 'material-ui/TextField';
import { FormControl } from 'material-ui/Form';
import Input, { InputLabel } from 'material-ui/Input';
import Select from 'material-ui/Select';
import { MenuItem } from 'material-ui/Menu';
import Button from 'material-ui/Button';
import Divider from 'material-ui/Divider';

import { DegreeType, EducationType } from '../../model';

/**
 * Props for EducationFields
 * @interface EducationFieldProps
 */
interface EducationFieldProps extends EducationType {
  handleChange: (event: React.FormEvent<{}>) => void;
  addEducation: (event: React.FormEvent<{}>) => void;
}

/**
 * EducationFields
 * @extends React.StatelessComponent<EducationFieldProps>
 */
const EducationFields: React.StatelessComponent<EducationFieldProps> = props => {
  return (
    <div>
      <div className="nabi-education">
        <TextField
          onChange={props.handleChange}
          fullWidth={true}
          id="schoolName"
          margin="normal"
          name="schoolName"
          placeholder="School name"
          required={true}
          value={props.schoolName}
        />
        <TextField
          onChange={props.handleChange}
          className="graduation-year"
          fullWidth={true}
          id="graduationYear"
          margin="normal"
          name="graduationYear"
          placeholder="Graduation year"
          required={true}
          value={props.graduationYear}
        />
        <TextField
          onChange={props.handleChange}
          fullWidth={true}
          id="fieldOfStudy"
          margin="normal"
          name="fieldOfStudy"
          placeholder="Field of study (major or concentration)"
          required={true}
          value={props.fieldOfStudy}
        />
        <FormControl>
          <InputLabel htmlFor="degreeType">Degree type</InputLabel>
          <Select
            input={<Input id="degreeType" name="degreeType" />}
            value={props.degreeType}
            onChange={props.handleChange}
          >
            <MenuItem value={DegreeType.associate}>{DegreeType.associate}</MenuItem>
            <MenuItem value={DegreeType.bachelors}>{DegreeType.bachelors}</MenuItem>
            <MenuItem value={DegreeType.graduate}>{DegreeType.graduate}</MenuItem>
            <MenuItem value={DegreeType.professional}>{DegreeType.professional}</MenuItem>
            <MenuItem value={DegreeType.certification}>{DegreeType.certification}</MenuItem>
            <MenuItem value={DegreeType.other}>{DegreeType.other}</MenuItem>
          </Select>
        </FormControl>
        <Button 
          color="primary"
          variant="raised"
          className="nabi-display-block nabi-margin-top-small"
          onClick={props.addEducation}
        >
          Add
        </Button>
      </div>
      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default EducationFields;
