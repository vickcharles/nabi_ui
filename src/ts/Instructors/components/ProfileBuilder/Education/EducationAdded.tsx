import * as React from 'react';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import Typography from 'material-ui/Typography';
import { EducationType }  from '../../../model';
import Divider from 'material-ui/Divider';

/**
 * Props for EducationAdded
 * @interface EducationAddedProps
 */
interface EducationAddedProps extends EducationType {
  deleteEducation: (educationId: any) => void;
}

/**
 * EducationAdded
 * @extends React.StatelessComponent<EducationAddedProps>
 */
const EducationAdded: React.StatelessComponent<EducationAddedProps> = props => {
  return (
    <div>
      <div className="selected-education">
        <IconButton className="nabi-float-right" aria-label="Delete" onClick={() => props.deleteEducation(props.id)}>
          <DeleteIcon />
        </IconButton>
        <Typography className="nabi-margin-top-small" variant="body2">
          {props.schoolName.toUpperCase()}
        </Typography>
        <Typography className="nabi-margin-top-small">
          {`${props.degreeType}, ${props.fieldOfStudy}`}
        </Typography>
        <Typography className="nabi-margin-top-small">
          {props.graduationYear}
        </Typography>
      </div>
      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default EducationAdded;
