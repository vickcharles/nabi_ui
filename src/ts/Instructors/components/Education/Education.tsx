import * as React from 'react';
import Typography from 'material-ui/Typography';
import Divider from 'material-ui/Divider';
// import { LessonDuration, RatesState } from '../../model';

import EducationFields from './EducationFields';

interface EducationProps {

}

const Education: React.StatelessComponent<EducationProps> = props => {
  return (
    <div>
     <Typography className="nabi-margin-top-small" variant="body2">
        Education
      </Typography>
      <EducationFields />
      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default Education;
