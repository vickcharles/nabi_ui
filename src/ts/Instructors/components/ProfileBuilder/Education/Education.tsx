import * as React from 'react';
import { Action, Dispatch } from 'redux';
import { connect } from 'react-redux';

import Typography from 'material-ui/Typography';

import EducationFields from './EducationFields';
import { DegreeType, EducationType, InstructorType } from '../../../model';
import { updateInstructor } from '../../../';
import EducationAdded from './EducationAdded';

/**
 * Education dispatch props
 * @interface EducationDispatchProps
 */
interface EducationDispatchProps {
  updateInstructor: (instructor: InstructorType) => void;
}

/**
 * Education own props
 * @interface EducationOwnProps
 */
interface EducationOwnProps {
  userId: string;
}

/**
 * Education props
 * @interface EducationProps
 * @extends BasicInfoDispatchProps & BasicInfoOwnProps
 */
interface EducationProps extends
  EducationDispatchProps,
  EducationOwnProps { }

/**
 * State for Education
 * @interface EducationState
 * @extends EducationType
 */
interface EducationState extends EducationType {
  education: EducationType[];
}

/**
 * Education: Second step on the Profile Builder
 * @class Education
 * @extends React.Component<EducationProps, EducationState>
 */
export class Education extends React.Component<EducationProps, EducationState > {
  constructor(props: EducationProps) {
    super(props);

    this.state = {
      id: 0,
      schoolName: '',
      graduationYear: '',
      degreeType: DegreeType.bachelors,
      fieldOfStudy: '',
      education: [],
    };

    this.handleChange = this.handleChange.bind(this);
    this.addEducation = this.addEducation.bind(this);
  }

  public handleChange(event: any): void {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      ...this.state,
      [name]: value
    });
  }

  public addEducation(): void {
    const educationToAdd: EducationType = {
      id: this.state.education.length,
      schoolName: this.state.schoolName,
      graduationYear: this.state.graduationYear,
      degreeType: this.state.degreeType,
      fieldOfStudy: this.state.fieldOfStudy
    };

    if (this.state.schoolName && this.state.graduationYear
      && this.state.degreeType && this.state.fieldOfStudy) {
      this.setState(
        { education: [...this.state.education, educationToAdd] },
        () => {
        const instructor: InstructorType =  {
          userId: this.props.userId,
          education: this.state.education
        };
        this.updateInstructorCall(instructor);
        this.clearEducation();
      });
    }
  }

  public updateInstructorCall(instructor: InstructorType): void {
    this.props.updateInstructor(instructor);
  }

  public clearEducation(): void {
    this.setState({  
      schoolName: '',
      graduationYear: '',
      degreeType: DegreeType.bachelors,
      fieldOfStudy: ''
    });
  }

  public deleteEducation(educationId: number): void {
    this.setState(
      { education: this.state.education.filter(
        education => education.id !== educationId)},
      () => {
      const instructor: InstructorType =  {
        userId: this.props.userId,
        education: this.state.education
      };
      this.updateInstructorCall(instructor);
    });
  }

  public render(): JSX.Element {
    const educationAdded = this.state.education.map((education, i) => (
      <EducationAdded
        key={i}
        id={education.id}
        schoolName={education.schoolName}
        graduationYear={education.graduationYear}
        degreeType={education.degreeType}
        fieldOfStudy={education.fieldOfStudy}
        deleteEducation={(educationId: number) => this.deleteEducation(educationId)}
      />
    ));
    return (
      <div>
        {educationAdded}
        <Typography className="nabi-margin-top-small" variant="body2">
          Add Education
        </Typography>
        <EducationFields 
          handleChange={this.handleChange} 
          addEducation={this.addEducation} 
          schoolName={this.state.schoolName}
          graduationYear={this.state.graduationYear}
          degreeType={this.state.degreeType}
          fieldOfStudy={this.state.fieldOfStudy}
        />
      </div>
    );
  }
}

function mapDispatchToProps(
  dispatch: Dispatch<Action>,
  _ownProps: EducationOwnProps
): EducationDispatchProps {
  return {
    updateInstructor: (instructor: InstructorType) => dispatch(updateInstructor(instructor)),
  };
}

export default connect(null, mapDispatchToProps)(Education);
