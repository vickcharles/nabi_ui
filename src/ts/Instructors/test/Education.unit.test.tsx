import * as React from 'react';
import { Education } from '../components/ProfileBuilder/Education/Education';
import { shallow } from 'enzyme';
import { InstructorType, DegreeType } from '../model';

describe('Education', () => {
  let wrapper: any;
  const updateInstructor: (instructor: InstructorType) => {} = jest.fn();
  const updateInstructorCall = jest.spyOn(Education.prototype, 'updateInstructorCall');
  const clearEducation = jest.spyOn(Education.prototype, 'clearEducation');

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Education userId="" updateInstructor={updateInstructor} />
      );
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Method handleChange()', () => {
    function test_handleChange(theName: string, theValue: any): void {
      describe(`When the event contains ${theName} for name and ${theValue} for value`, () => {
        beforeEach(() => {
          const e = {
            target: {
              name: theName,
              value: theValue
            }
          };

          wrapper.instance().handleChange(e);
        });

        it(`Sets the state's ${theName} to ${theValue}`, () => {
          expect(wrapper.state(theName)).toBe(theValue);
        });
      });
    }

    test_handleChange('schoolName', 'UMass Boston');
    test_handleChange('graduationYear', 2012);
  });

  describe('Method addEducation()', () => {
    describe('When id, schoolName, graduationYear, degreeType and fieldStudy state are missing', () => {
      beforeEach(() => {
        wrapper.instance().addEducation();
      });

      it('Does not add education()', () => {
        expect(wrapper.state('education').length).toBe(0);
      });

      it('Does not call updateInstructorCall', () => {
        expect(updateInstructorCall).not.toBeCalled();
      });
    });

    describe('When id, schoolName, graduationYear, degreeType and fieldStudy state are defined', () => {
      beforeEach(() => {
        wrapper.setState({
          schoolName: 'UmassBoston',
          graduationYear: '2012',
          degreeType: DegreeType.bachelors,
          fieldOfStudy: 'Music',
        });
        wrapper.instance().addEducation();
      });

      it('Adds education()', () => {
        expect(wrapper.state('education').length).toBe(1);
      });

      it('Calls updateInstructorCall', () => {
        expect(updateInstructorCall).toBeCalled();
      });

      it('Calls clearEducation', () => {
        expect(clearEducation).toBeCalled();
      });
    });
  });

  describe('Method clearEducation()', () => {
    beforeEach(() => {
      wrapper.instance().clearEducation();
    });

    it('Clears education', () => {
      expect(wrapper.state('schoolName')).toBe('');
      expect(wrapper.state('graduationYear')).toBe('');
      expect(wrapper.state('degreeType')).toBe(DegreeType.bachelors);
      expect(wrapper.state('fieldOfStudy')).toBe('');
    });
  });

  describe('Method deleteEducation()', () => {
    beforeEach(() => {
      wrapper.instance().deleteEducation();
    });

    it('Calls updateInstructorCall', () => {
      expect(updateInstructorCall).toBeCalled();
    });
  });
});
