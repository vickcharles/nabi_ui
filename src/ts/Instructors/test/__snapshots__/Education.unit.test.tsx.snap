// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education Shallow Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Education
    updateInstructor={[MockFunction]}
    userId=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <WithStyles(Typography)
          className="nabi-margin-top-small"
          variant="body2"
        >
          Add Education
        </WithStyles(Typography)>,
        <EducationFields
          addEducation={[Function]}
          degreeType="Bachelor's Degree"
          fieldOfStudy=""
          graduationYear=""
          handleChange={[Function]}
          schoolName=""
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Add Education",
          "className": "nabi-margin-top-small",
          "variant": "body2",
        },
        "ref": null,
        "rendered": "Add Education",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addEducation": [Function],
          "degreeType": "Bachelor's Degree",
          "fieldOfStudy": "",
          "graduationYear": "",
          "handleChange": [Function],
          "schoolName": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <WithStyles(Typography)
            className="nabi-margin-top-small"
            variant="body2"
          >
            Add Education
          </WithStyles(Typography)>,
          <EducationFields
            addEducation={[Function]}
            degreeType="Bachelor's Degree"
            fieldOfStudy=""
            graduationYear=""
            handleChange={[Function]}
            schoolName=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add Education",
            "className": "nabi-margin-top-small",
            "variant": "body2",
          },
          "ref": null,
          "rendered": "Add Education",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addEducation": [Function],
            "degreeType": "Bachelor's Degree",
            "fieldOfStudy": "",
            "graduationYear": "",
            "handleChange": [Function],
            "schoolName": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
